File: JAVA_COURSE/KOTLIN/09_Robot2/src/JavaRobot/RobotExt.java
package JavaRobot;

public class RobotExt extends Robot
{
    public RobotExt(double x, double y, double course) {
        super(x, y);
        this.course = course;
    }

    public void back(int distance) {
        forward(-distance);
    }
     // Печать координат робота
    public void printCoordinates() {
        System.out.println(getX() + "," + getY() + "Это я переопределил метод");
    }
}

File: JAVA_COURSE/KOTLIN/09_Robot2/src/kotlinrobot/RobotExt.kt
package kotlinrobot

class RobotExt(x: Double, y: Double, course: Double) : Robot(x, y) {
    init {
        this.course = course
    }

    fun back(distance: Int) {
        forward(-distance)
    }

    // Печать координат робота
    override fun printCoordinates() {
        println(x.toString() + "," + y + "Это я переопределил метод")
    }
}
